test/integration/scheduler/scheduler_test.go

```diff
@@ -77,593 +72,460 @@ var _ = ginkgo.Describe("Scheduler", func() {
-		gomega.Expect(k8sClient.Create(ctx, devBEQueue)).Should(gomega.Succeed())
-
-		wl1 := testing.MakeWorkload("wl-1", ns.Name).Queue(prodBEQueue.Name).Request(corev1.ResourceCPU, "11").Obj()
-		wl2 := testing.MakeWorkload("wl-2", ns.Name).Queue(devBEQueue.Name).Request(corev1.ResourceCPU, "11").Obj()
-
-		ginkgo.By("Creating two workloads")
-		gomega.Expect(k8sClient.Create(ctx, wl1)).Should(gomega.Succeed())
-		gomega.Expect(k8sClient.Create(ctx, wl2)).Should(gomega.Succeed())
-		framework.ExpectWorkloadsToBePending(ctx, k8sClient, wl1, wl2)
-		framework.ExpectPendingWorkloadsMetric(prodBEQueue, 1)
-		framework.ExpectPendingWorkloadsMetric(devBEQueue, 1)
-
-		// Make sure workloads are in the same scheduling cycle.
-		testBEClusterQ := testing.MakeClusterQueue("test-be-cq").
-			Cohort("be").
-			Resource(testing.MakeResource(corev1.ResourceCPU).
-				Flavor(testing.MakeFlavor(onDemandFlavor.Name,
-					"15").Max("15").Obj()).
-				Obj()).
-			Obj()
-		gomega.Expect(k8sClient.Create(ctx, testBEClusterQ)).Should(gomega.Succeed())
-		defer func() {
-			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, prodBEClusterQ)).Should(gomega.Succeed())
-			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, devBEClusterQ)).Should(gomega.Succeed())
-			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, testBEClusterQ)).Should(gomega.Succeed())
-		}()
-
-		framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodBEClusterQ.Name, wl1)
-		framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, devBEClusterQ.Name, wl2)
-		framework.ExpectPendingWorkloadsMetric(prodBEQueue, 0)
-		framework.ExpectPendingWorkloadsMetric(devBEQueue, 0)
+	ginkgo.When("Handing clusterQueue events", func() {
+		var (
+			cq    *kueue.ClusterQueue
+			queue *kueue.Queue
+		)
+
+		ginkgo.BeforeEach(func() {
+			gomega.Expect(k8sClient.Create(ctx, onDemandFlavor)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, spotTaintedFlavor)).Should(gomega.Succeed())
+
+			cq = testing.MakeClusterQueue("cluster-queue").
+				Cohort("be").
+				Resource(testing.MakeResource(corev1.ResourceCPU).
+					Flavor(testing.MakeFlavor(spotTaintedFlavor.Name, "5").Max("5").Obj()).
+					Flavor(testing.MakeFlavor(onDemandFlavor.Name, "5").Obj()).
+					Obj()).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, cq)).Should(gomega.Succeed())
+			queue = testing.MakeQueue("queue", ns.Name).ClusterQueue(cq.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, queue)).Should(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, cq)).To(gomega.Succeed())
+			gomega.Expect(framework.DeleteResourceFlavor(ctx, k8sClient, onDemandFlavor)).To(gomega.Succeed())
+			gomega.Expect(framework.DeleteResourceFlavor(ctx, k8sClient, spotTaintedFlavor)).To(gomega.Succeed())
+		})
+		ginkgo.It("Should re-enqueue by the update event of ClusterQueue", func() {
+			fooCq := testing.MakeClusterQueue("foo-clusterqueue").
```

Comment:
move this to the BeforeEach and remove the `cq` one, it is not used for this test.

Reply:
We actually used below.

Reply:
Sorry, we create three jobs in this testcase, job1 and job3 with the prodQueue, job2 with the devBEQueue, they belong to different cqs. https://github.com/kubernetes-sigs/kueue/blob/cb409f6562a9222c966bb08aeda4d6ae3c64f0e2/test/integration/scheduler/scheduler_test.go#L493-L515

But I think we can actually test the update event with one clusterQueue, but I have no idea what's the original test intension.

Reply:
ah, right. As you mentioned, this can be tested using a single CQ, I think this was just copied from the other test "Should re-enqueue by the delete event of workload belonging to the same Cohort" and modified.

We can remove job1 and job3 and their cq.

Reply:
updated.